delete.js

```diff
@@ -181,83 +231,139 @@ async function getHarvestedFiles(uri) {
  * @param {string} submissionDocument URI of the submitted document to get the related file for
  * @param {string} fileType URI of the type of the related file
  */
-async function getTTLResource(submissionDocument, fileType) {
+async function getTTLResource(submissionDocument, fileType, graph) {
   const response = await querySudo(`
     PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
     PREFIX dct: <http://purl.org/dc/terms/>
     PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
 
-    SELECT DISTINCT ?file
+    SELECT DISTINCT ?logical ?physical
     WHERE {
-      GRAPH ?g {
-        ${sparqlEscapeUri(submissionDocument)} dct:source ?file .
-      }
-      GRAPH ${sparqlEscapeUri(FILE_GRAPH)} {
-        ?file dct:type ${sparqlEscapeUri(fileType)} .
+      GRAPH ${sparqlEscapeUri(graph)} {
+        ${sparqlEscapeUri(submissionDocument)} dct:source ?physical .
+        ?physical
+          dct:type ${sparqlEscapeUri(fileType)} ;
+          nie:dataSource ?logical .
       }
     } LIMIT 1
   `);
 
   const parser = new SparqlJsonParser();
   const parsedResults = parser.parseJsonResults(response);
-  if (parsedResults.length < 1)
+  if (parsedResults.length < 1) {
     console.log(
       `Part of type ${fileType} for submission document ${submissionDocument} not found`,
     );
-  return parsedResults[0]?.file?.value;
+    return;
+  }
+  return {
+    logical: parsedResults[0]?.logical?.value,
+    physical: parsedResults[0]?.physical?.value,
+  };
 }
 
 /**
  * Delete submission resource
  *
  * @param {string} URI of the resource to delete the related files for
  */
-async function deleteResource(URI) {
-  return querySudo(`
+async function deleteResource(uri, graph) {
+  return updateSudo(`
     DELETE {
-      GRAPH ?g {
-        ${sparqlEscapeUri(URI)} ?p ?o .
+      GRAPH ${sparqlEscapeUri(graph)} {
+        ${sparqlEscapeUri(uri)} ?p ?o .
       }
     }
     WHERE {
-      GRAPH ?g {
-        ${sparqlEscapeUri(URI)} ?p ?o .
+      GRAPH ${sparqlEscapeUri(graph)} {
+        ${sparqlEscapeUri(uri)} ?p ?o .
       }
     }
   `);
 }
 
-async function getSubmissionById(submissionId) {
+async function deleteTaskwithJob(taskUri, graph) {
+  return updateSudo(`
+    PREFIX dct: <http://purl.org/dc/terms/>
+    PREFIX task: <http://redpencil.data.gift/vocabularies/tasks/>
+
+    DELETE {
+      GRAPH ${sparqlEscapeUri(graph)} {
+        ?hc2 ?hp2 ?ho2 .
+        ?hc ?hp ?ho .
+        ?rc ?rp ?ro .
+        ?ic ?ip ?io .
+        ?task ?tp ?to .
+        ?job ?jp ?jo .
+      }
+    } WHERE {
+      GRAPH ${sparqlEscapeUri(graph)} {
+        ${sparqlEscapeUri(taskUri)} dct:isPartOf ?job .
```

Comment:
is `${sparqlEscapeUri(taskUri)}` the same as the `?task` in the latter part of the query? wouldn't be a bind better?